### PR TITLE
Chart Refactor

### PR DESCRIPTION
**Pull request by Drayz64**

### CODE REVIEW

File: src/main/java/tempest/ui/ViewManager.java [hunk 1]
```diff
@@ -1,98 +1,100 @@
...
-    return getView(classObject).getName();
-  }
+    /**
+     * Finds the name of the first instance of a particular class in the possible
+     * views.
+     *
+     * @param classObject The class to be found.
+     * @return The view, if it exists; else null.
+     */
+    public String getViewName(Class<? extends Page> classObject) {
```
Comment (Drayz64) [inline src/main/java/tempest/ui/ViewManager.java:97]: Can this be deleted? it isn't used by anything at the moment

File: src/main/java/tempest/ui/components/BackButton.java [hunk 2]
```diff
@@ -19,15 +17,4 @@ public BackButton(GUIManager manager) {
...
         this.addActionListener(e -> manager.swapToPrevCard());
     }
-
-    /**
-     * Navigation button that sends the user back to the previous chart.
-     * 
-     * @param manager The instance of manager passed into the page class
-     */
-    public BackButton(ViewManager<Chart> manager) {
-        super("Back");
```
Comment (bridges-wood) [inline src/main/java/tempest/ui/components/BackButton.java:29]: Why is the name of the button not set in the new implementation?

File: src/main/java/tempest/ui/pages/InputPage.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package tempest.ui.pages;
+
+/**
+ * Pages should implement this interface if they want to use a
+ * clear button
+ */
+public interface InputPage {
+    // Used to clear input fields when the clear button is pressed
```
Comment (bridges-wood) [inline src/main/java/tempest/ui/pages/InputPage.java:8]: Could you change this to JavaDoc format?

**Review by bridges-wood — CHANGES_REQUESTED** [review]
Noice, just a minor change and a question.

File: src/main/java/tempest/ui/components/BackButton.java (hunk 2 above)
Comment (Drayz64) [reply src/main/java/tempest/ui/components/BackButton.java:29]: It is in the method above _(had to extend the code to see it)_

```java
public BackButton(GUIManager manager) {
    super("Back");
    this.setFocusable(false);
    this.addActionListener(e -> manager.swapToPrevCard());
}
```

That ViewManager<Chart> one wasn't used by anything

**Review by bridges-wood — APPROVED** [review]

**Pull request merged**